arwen/elrondapi/bigFloatOps.go

```diff
@@ -0,0 +1,893 @@
+
+// // Declare the function signatures (see [cgo](https://golang.org/cmd/cgo/)).
+//
+// #include <stdlib.h>
+// typedef unsigned char uint8_t;
+// typedef int int32_t;
+//
+// extern int32_t	v1_4_bigFloatNew(void* context, int32_t intBase, int32_t subIntBase, int32_t exponent);
+// extern int32_t	v1_4_bigFloatNewFromFrac(void* context, long long numerator, long long denominator);
+//
+// extern void		v1_4_bigFloatAdd(void* context, int32_t destinationHandle, int32_t op1Handle, int32_t op2Handle);
+// extern void		v1_4_bigFloatSub(void* context, int32_t destinationHandle, int32_t op1Handle, int32_t op2Handle);
+// extern void		v1_4_bigFloatMul(void* context, int32_t destinationHandle, int32_t op1Handle, int32_t op2Handle);
+// extern void		v1_4_bigFloatDiv(void* context, int32_t destinationHandle, int32_t op1Handle, int32_t op2Handle);
+// extern void		v1_4_bigFloatRoundDiv(void* context, int32_t destinationHandle, int32_t op1Handle, int32_t op2Handle);
+// extern void		v1_4_bigFloatMod(void* context, int32_t destinationHandle, int32_t op1Handle, int32_t op2Handle);
+//
+// extern void		v1_4_bigFloatAbs(void* context, int32_t destinationHandle, int32_t opHandle);
+// extern void		v1_4_bigFloatNeg(void* context, int32_t destinationHandle, int32_t opHandle);
+// extern int32_t	v1_4_bigFloatCmp(void* context, int32_t op1Handle, int32_t op2Handle);
+// extern int32_t	v1_4_bigFloatSign(void* context, int32_t opHandle);
+// extern void 		v1_4_bigFloatCopy(void* context, int32_t destinationHandle, int32_t opHandle);
+// extern void		v1_4_bigFloatSqrt(void* context, int32_t destinationHandle, int32_t opHandle);
+// extern int32_t	v1_4_bigFloatLog2(void* context, int32_t opHandle);
+// extern void		v1_4_bigFloatPow(void* context, int32_t destinationHandle, int32_t op1Handle, int32_t op2Handle);
+//
+// extern void		v1_4_bigFloatFloor(void* context, int32_t opHandle, int32_t bigIntHandle);
+// extern void		v1_4_bigFloatCeil(void* context, int32_t opHandle, int32_t bigIntHandle);
+//
+// extern int32_t	v1_4_bigFloatIsInt(void* context, int32_t opHandle);
+// extern void		v1_4_bigFloatSetInt64(void* context, int32_t destinationHandle, long long value);
+// extern void		v1_4_bigFloatSetBigInt(void* context, int32_t destinationHandle, int32_t bigIntHandle);
+//
+// extern void		v1_4_bigFloatGetConstPi(void* context, int32_t destinationHandle);
+// extern void		v1_4_bigFloatGetConstE(void* context, int32_t destinationHandle);
+//
+// extern void		v1_4_bigFloatSetBytes(void* context, int32_t destinationHandle, int32_t dataOffset, int32_t dataLength);
+// extern void		v1_4_bigFloatGetBytes(void* context, int32_t destinationHandle, int32_t dataOffset);
+// extern void		v1_4_bigFloatFinish(void* context, int32_t referenceHandle);
+// extern void		v1_4_bigFloatGetArgument(void* context, int32_t id, int32_t destinationHandle);
+//
+// extern int32_t	v1_4_bigFloatStorageStore(void* context, int32_t keyOffset, int32_t keyLength, int32_t sourceHandle);
```

Comment:
Implement conversions to/from managed buffers, after that specialized storage/argument/result hooks are no longer needed.